megamek/src/megamek/client/ui/swing/dialog/AbstractUnitSelectorDialog.java

```diff
@@ -0,0 +1,934 @@
+    protected JButton buttonSelect;
+    protected JButton buttonClose;
+    protected JButton buttonShowBV;
+
+    private JButton buttonAdvancedSearch;
+    private JButton buttonResetSearch;
+    protected JList<String> listTechLevel = new JList<>();
+    /**
+     * We need to map the selected index of listTechLevel to the actual TL it
+     * belongs to
+     */
+    protected Map<Integer, Integer> techLevelListToIndex = new HashMap<>();
+    protected JComboBox<String> comboUnitType = new JComboBox<>();
+    protected JComboBox<String> comboWeight = new JComboBox<>();
+    protected JLabel labelImage = new JLabel(""); //inline to avoid potential null pointer issues
+    protected JTable tableUnits;
+    protected JTextField textFilter;
+    private MechViewPanel panelMechView;
+    private MechViewPanel panelTROView;
+
+    private StringBuffer searchBuffer = new StringBuffer();
+    private long lastSearch = 0;
+    // how long after a key is typed does a new search begin
+    private static final int KEY_TIMEOUT = 1000;
+
+    protected static MechSummaryCache mscInstance = MechSummaryCache.getInstance();
+    protected MechSummary[] mechs;
+
+    private MechTableModel unitModel = new MechTableModel();
+    protected MechSearchFilter searchFilter;
+
+    private UnitLoadingDialog unitLoadingDialog;
+    private AdvancedSearchDialog asd;
+    protected JFrame frame;
+
+    protected TableRowSorter<MechTableModel> sorter;
+
+    protected GameOptions gameOptions = null;
+    protected boolean enableYearLimits = false;
+    protected int allowedYear = ALLOWED_YEAR_ANY;
+    protected boolean canonOnly = false;
+    protected int gameTechLevel = TechConstants.T_SIMPLE_INTRO;
+    protected int techLevelDisplayType = TECH_LEVEL_DISPLAY_IS_CLAN;
+
+    protected static MMLogger logger = DefaultMmLogger.getInstance();
+    //endregion Variable Declarations
+
+    protected AbstractUnitSelectorDialog(JFrame frame, UnitLoadingDialog unitLoadingDialog) {
+        super(frame, Messages.getString("MechSelectorDialog.title"), true); //$NON-NLS-1$
+        setName("UnitSelectorDialog");
+        this.frame = frame;
+        this.unitLoadingDialog = unitLoadingDialog;
+        super.setVisible(false);
+    }
+
+    /**
+     * This is used to update any values that are set based on individual options
+     */
+    public abstract void updateOptionValues();
+
+    /**
+     * This has been set up to permit preference implementation in anything that extends this
+     */
+    private void setUserPreferences() {
+        GUIPreferences guiPreferences = GUIPreferences.getInstance();
+        setSize(guiPreferences.getMechSelectorSizeWidth(), guiPreferences.getMechSelectorSizeHeight());
+
+        comboUnitType.setSelectedIndex(guiPreferences.getMechSelectorUnitType());
+
+        comboWeight.setSelectedIndex(guiPreferences.getMechSelectorWeightClass());
+
+        updateTypeCombo(guiPreferences.getMechSelectorRulesLevels().replaceAll("\\[", ""));
```

Comment:
Why is this replaced here and not below where the right bracket is replaced?

Reply:
Yep, inherited code... but I'm not sure.

I've reworked it so that it just sends the guiPreferences to updateTypeCombo and if needed it gets the rules and does the replacements, as that's just a better way to handle it.